Assets/Scripts/DaggerfallUnityApplication.cs

```diff
@@ -0,0 +1,107 @@
+// Project:         Daggerfall Unity
+// Copyright:       Copyright (C) 2009-2024 Daggerfall Workshop
+// Web Site:        http://www.dfworkshop.net
+// License:         MIT License (http://www.opensource.org/licenses/mit-license.php)
+// Source Code:     https://github.com/Interkarma/daggerfall-unity
+// Original Author: kaboissonneault
+// Contributors:    
+// 
+// Notes:
+//
+
+//#define SEPARATE_DEV_PERSISTENT_PATH
+
+using System;
+using System.IO;
+using UnityEngine;
+
+public static class DaggerfallUnityApplication
+{
+    static string persistentDataPath;
+    public static string PersistentDataPath { get { return persistentDataPath; } }
+
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+    static void SubsystemInit()
+    {
+#if UNITY_EDITOR && SEPARATE_DEV_PERSISTENT_PATH
+        persistentDataPath = String.Concat(Application.persistentDataPath, ".devenv");
+        Directory.CreateDirectory(persistentDataPath);
+#else
+        persistentDataPath = Application.persistentDataPath;
+#endif
+
+        InitLog();
+    }
+
+
+    public class LogHandler : ILogHandler, IDisposable
+    {
+        private FileStream fileStream;
+        private StreamWriter streamWriter;
+
+        public LogHandler()
+        {
+            string filePath = Path.Combine(PersistentDataPath, "Player.log");
+
+            try
+            {
+                if(File.Exists(filePath))
+                {
+                    string prevPath = Path.Combine(PersistentDataPath, "Player-prev.log");
+                    File.Delete(prevPath);
+                    File.Move(filePath, prevPath);
+                }
+            }
+            catch { }
+
+            fileStream = new FileStream(filePath, FileMode.Create, FileAccess.ReadWrite);
+            streamWriter = new StreamWriter(fileStream);
+        }
+
+        public void LogException(Exception exception, UnityEngine.Object context)
+        {
+            streamWriter.WriteLine(exception.ToString());
+            streamWriter.Flush();
```

Comment:
If you're planning on flushing after every write, you could set `StreamWriter.AutoFlush` to `true`. That would ensure consistent behavior in the future if somebody else updates the type.